tests/tests/test-precompile/test-precompile-xcm-utils.ts

```diff
@@ -0,0 +1,123 @@
+import "@moonbeam-network/api-augment";
+
+import { u8aToHex } from "@polkadot/util";
+import { expect } from "chai";
+import { ethers } from "ethers";
+import { PRECOMPILE_XCM_UTILS_ADDRESS } from "../../util/constants";
+import { getCompiled } from "../../util/contracts";
+import { web3EthCall } from "../../util/providers";
+import { describeDevMoonbeamAllEthTxTypes, DevTestContext } from "../../util/setup-dev-tests";
+
+import { descendOriginFromAddress } from "../../util/xcm";
+
+const XCM_UTILS_CONTRACT = getCompiled("XcmUtils");
+const XCM_UTILSTRANSACTOR_INTERFACE = new ethers.utils.Interface(XCM_UTILS_CONTRACT.contract.abi);
+
+describeDevMoonbeamAllEthTxTypes("Precompiles - xcm UTILS", (context) => {
+  it("allows to retrieve parent-based ML account", async function () {
+    const multilocation: [number, {}[]] =
+      // Destination as multilocation
+      [
+        // one parent
+        1,
+        [],
+      ];
+    let result = await web3EthCall(context.web3, {
+      to: PRECOMPILE_XCM_UTILS_ADDRESS,
+      data: XCM_UTILSTRANSACTOR_INTERFACE.encodeFunctionData("multilocationToAddress", [
+        multilocation,
+      ]),
+    });
+
+    const expectedAddress = u8aToHex(new Uint8Array([...new TextEncoder().encode("Parent")]))
+      .slice(2)
```

Comment:
optional: `u8aToHex(new Uint8Array([...new TextEncoder().encode("Parent")], null, false)` will output the hex without `0x` prefix, avoiding `slice(2)`. The `null` is the bitLength parameter and `false` dictates the prefix.